rmw_dds_common/src/security.cpp

```diff
@@ -40,20 +40,27 @@ bool get_security_files(
   };
 
   for (const std::pair<const std::string, std::string> & el : required_files) {
-    rcpputils::fs::path full_path(secure_root);
+    std::filesystem::path full_path(secure_root);
     full_path /= el.second;
-    if (!full_path.is_regular_file()) {
+    std::filesystem::directory_entry target(full_path);
+    if (!target.is_regular_file()) {
       result.clear();
       return false;
     }
 
     result[el.first] = prefix + full_path.string();
   }
 
+  // Check all required files are exists
+  if (result.size() != required_files.size()) {
+    return false;
+  }
```

Comment:
I don't think that this check is necessary.  If any of them were missing, the `for` loop above would have returned `false` already.

Reply:
Oh.. yes.
I was confused. I will remove the loop.
Thanks!